# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/zstd/hotfix-cross-no-gen_html.patch
# Copyright (C) 2026 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

gen_html is a build-time tool executed on the build host to generate
zstd_manual.html; when cross-compiling it is built for the target and
thus "not executable" on the host.

--- zstd-1.5.7/build/cmake/contrib/CMakeLists.txt
+++ zstd-1.5.7/build/cmake/contrib/CMakeLists.txt
@@ -10,4 +10,7 @@
 project(contrib)
 
 add_subdirectory(pzstd)
-add_subdirectory(gen_html)
+# gen_html runs on the build host, cannot execute when cross-compiling
+if(NOT CMAKE_CROSSCOMPILING)
+    add_subdirectory(gen_html)
+endif()
